libs/domain/product/src/services/adapter/normalizers/product/product.normalizer.ts

```diff
@@ -129,6 +128,41 @@ export function productCategoryNormalizer(
   return transformer.transform(node, CategoryNormalizer);
 }
 
+export function productVariantNormalizer(
+  data: DeserializedProduct
+): Partial<Product> {
+  const { abstractProducts } = data;
+
+  const abstract = abstractProducts?.[0];
+
+  const variantMap = abstract.attributeMap?.attributeVariantMap;
+  const variantDefinition = abstract.attributeMap?.superAttributes;
+
+  if (variantMap) {
+    const variantMapEntries = Object.entries(variantMap);
+
+    const variants = variantMapEntries.reduce((skuMap, [_, mapAttributes]) => {
+      const matchingVariant = abstract.concreteProducts.find(
+        (variant) =>
+          variant.attributes &&
+          Object.keys(mapAttributes).every(
+            (attrKey) => variant.attributes![attrKey] === mapAttributes[attrKey]
+          )
+      );
+
+      if (matchingVariant && matchingVariant.sku) {
+        skuMap[matchingVariant.sku] = mapAttributes;
+      }
+
+      return skuMap;
+    }, {} as Record<string, Record<string, string>>);
+
+    return { variants, variantDefinition };
```

Comment:
If we providing parsing anywaay can we just move specifiv variantDefinition into variants object?